packages/eslint/src/plugins/plugin.ts

```diff
@@ -28,48 +29,72 @@ export const createNodes: CreateNodes<EslintPluginOptions> = [
     baseEsLintConfigFile,
     baseEsLintFlatConfigFile,
   ]),
-  (configFilePath, options, context) => {
+  async (configFilePath, options, context) => {
     options = normalizeOptions(options);
 
     // Ensure that configFiles are set, e2e-run fails due to them being undefined in CI (does not occur locally)
     // TODO(JamesHenry): Further troubleshoot this in CI
     (context as any).configFiles = context.configFiles ?? [];
 
-    // Create a Set of all the directories containing eslint configs
-    const eslintRoots = new Set(context.configFiles.map(dirname));
+    // Create a Set of all the directories containing eslint configs, and a
+    // list of globs to exclude from child projects
+    const eslintRoots = new Set();
+    const filesToExclude: string[] = [];
+    for (const configFile of context.configFiles) {
+      const configDir = dirname(configFile);
+      eslintRoots.add(configDir);
+      if (configDir !== '.' && configDir !== dirname(configFilePath)) {
+        filesToExclude.push(`${configDir}/**/*`);
+      }
+    }
+
     const configDir = dirname(configFilePath);
+    const childProjectRoots = new Set<string>();
+    const ESLint = resolveESLintClass(isFlatConfig(configFilePath));
 
-    const childProjectRoots = globWithWorkspaceContext(
+    const projectFiles = globWithWorkspaceContext(
       context.workspaceRoot,
       [
         'project.json',
         'package.json',
         '**/project.json',
         '**/package.json',
       ].map((f) => join(configDir, f))
-    )
-      .map((f) => dirname(f))
-      .filter((childProjectRoot) => {
-        // Filter out projects under other eslint configs
-        let root = childProjectRoot;
-        // Traverse up from the childProjectRoot to either the workspaceRoot or the dir of this config file
-        while (root !== dirname(root) && root !== dirname(configFilePath)) {
-          if (eslintRoots.has(root)) {
-            return false;
-          }
-          root = dirname(root);
+    );
+    projectFileLoop: for (const projectFile of projectFiles) {
+      const childProjectRoot = dirname(projectFile);
+      if (childProjectRoots.has(childProjectRoot)) {
+        continue;
+      }
+
+      // Filter out projects under other eslint configs
+      let root = childProjectRoot;
+      // Traverse up from the childProjectRoot to either the workspaceRoot or the dir of this config file
+      while (root !== dirname(root) && root !== dirname(configFilePath)) {
+        if (eslintRoots.has(root)) {
+          continue projectFileLoop;
         }
-        return true;
-      })
-      .filter((dir) => {
-        // Ignore project roots where the project does not contain any lintable files
-        const lintableFiles = globWithWorkspaceContext(context.workspaceRoot, [
-          join(dir, `**/*.{${options.extensions.join(',')}}`),
-        ]);
-        return lintableFiles.length > 0;
```

Comment:
So this never handled if there were lintable files.. but they were all under a child config path right?

Reply:
Exactly. It never handled the ESLint ignored file configuration. It did glob for children with specific file extensions, which might or might not have been lintable for the project.